Move GL interface code out of the OpenGL video backend.

phire opened the pull request:
This moves the OpenGL loading/interface/context-creation code out of the OpenGL video backend into Common/GL. This means Video Software no-longer depends on the OpenGL backend being linked in to function, a dependency that made little sense.

The reason it was moved into Common/GL rather than something like VideoCommon/GL is that VideoCommon is for code common to both the OpenGL and DirectX11 backends, which GL context creation is clearly not. It is our intention in the future to potentially rename VideoCommon to something like FlipperCommon and limit it's contents to common code for the emulation the GameCube's GPU.

A longer-term goal of is to allow the creation of an OpenGL based "Big Picture" user interface that will create a OpenGL context as soon as Dolphin starts, without having to initialize the OpenGL backend first.

degasus commented on Source/Core/Common/GL/GLInterface/EGL.cpp:
```diff
@@ -2,8 +2,8 @@
-#include "VideoBackends/OGL/GLInterface/EGL.h"
+#include "Common/GL/GLInterfaceBase.h"
+#include "Common/GL/GLInterface/EGL.h"
 #include "VideoCommon/DriverDetails.h"
```
Common must not depend on VideoCommon, else linking will fail.

degasus commented on Source/Core/Common/GL/GLInterface/EGL.h:
```diff
@@ -7,8 +7,8 @@
 #include <EGL/egl.h>
 
+#include "Common/GL/GLInterfaceBase.h"
 #include "Core/ConfigManager.h"
```
Neither on Core

phire replied:
I'll fix those in my next commit.

phire commented:
This moves the OpenGL loading/interface/context-creation code out of the OpenGL video backend into `Common/GL`. This means Video Software no-longer depends on the OpenGL backend being linked in to function, a dependency that made little sense. 

The reason it was moved into `Common/GL` rather than something like `VideoCommon/GL` is that VideoCommon is for code common to both the OpenGL and DirectX11 backends, which GL context creation is clearly not. It is our intention in the future to potentially rename VideoCommon to something like FlipperCommon and limit it's contents to common code for the emulation the GameCube's GPU. 

A longer-term goal of is to allow the creation of an OpenGL based "BigPicture" user interface that will create a OpenGL context as soon as Dolphin starts, without having to initialize the OpenGL backend first.

degasus commented on Source/Core/Common/GL/GLInterface/EGLAndroid.cpp:
```diff
@@ -2,8 +2,8 @@
 // Refer to the license.txt file included.
 
+#include "Common/GL/GLInterface/EGLAndroid.h"
 #include "Core/Host.h"
```
You're also not allowed to use Core.

degasus replied:
Core dependency

degasus commented on Source/Core/VideoBackends/OGL/Render.cpp:
```diff
@@ -24,8 +26,7 @@
 #include "VideoBackends/OGL/FramebufferManager.h"
-#include "VideoBackends/OGL/GLInterfaceBase.h"
-#include "VideoBackends/OGL/GLUtil.h"
+
```
No need for an empty line.

phire commented:
I've removed the final stray include and fixed up a few newline issues. 

Only two of those dependencies on VideoCommon were real, all other dependencies were just stray includes.

waddlesplash commented:
LGTM
